csrc/scheduler/registry_utils.cpp

```diff
@@ -614,20 +586,111 @@ bool SchedulerTopologyChecker::hasNonNormalizePostReductionBCast(
         // propagated as we map the IDs through the backward traversal.
         std::vector<std::pair<IterDomain*, IterDomain*>> ids_to_resolve;
 
-        // Check if any TensorViews have a resolved broadcast
-        for (auto entry : forward_p2c_root_map) {
-          auto p_id = entry.first;
-          auto c_id = entry.second;
-          if (p_id->isBroadcast() && !c_id->isBroadcast()) {
-            ids_to_resolve.emplace_back(c_id, c_id);
+        // (1) step-1, Find the id to resolve
+        while (!forward_tv_dep_chain.empty()) {
+          forward_running_producer = forward_running_consumer;
+          forward_running_consumer = forward_tv_dep_chain.front();
+          forward_tv_dep_chain.pop_front();
+
+          if (std::none_of(
+                  forward_running_producer->getLogicalDomain().begin(),
+                  forward_running_producer->getLogicalDomain().end(),
+                  [](IterDomain* id) { return id->isBroadcast(); })) {
+            // If there's no broadcast axes in producer it doesn't need to be
+            // checked
+            continue;
+          }
+
+          // If consumer is before another reduction it doesn't need to be
+          // checked
+          if (pre_reduction_tvs.count(forward_running_consumer)) {
```

Comment:
When does this happen?

Reply:
Previously, when this condition hits, it exits the while loop. In this version, it just exits the inner while loop and moves on to the step 2. What does the difference mean?

Reply:
> When does this happen?

when we have two consecutive reductions, e.g. softmax we have
```
|-------->------|
0 -> r1 -> b2 ->3 ->4 -> r5
```
where tensor `3` is an input to reduction `r5`, since the compile time check ensures `r1` and `r5` are same, so the check is deferred to r5's consumers.

Reply:
> Previously, when this condition hits, it exits the while loop. In this version, it just exits the inner while loop and moves on to the step 2. What does the difference mean?

before going to step-2, it is captured by further checks of ids_to_resolve, see [code](https://github.com/NVIDIA/Fuser/blob/3153d3ae74edd6902a63af9f5c6d1e7c1aec1ccb/csrc/scheduler/registry_utils.cpp#L638).
```
        // break from this chain if no id to resolve
        if (ids_to_resolve.empty()) {
          break;
        }
```
and breaks from this chain, so still same to the original version.

Added comments for clarity.

Reply:
I see. Makes sense.